### PR TITLE
Add filter by column for coach and hr

### PR DESCRIPTION
**Pull request by ExxZzyy**
Add filter by column for coach and hr
image

### CODE REVIEW

File: src/app/models/hr-profile/hr-profile.component.ts [hunk 1]
```diff
@@ -24,20 +39,50 @@ export class HrProfileComponent implements AfterViewInit {
...
+      this.filterValues.firstName = firstName;
+      this.dataSource.filter = JSON.stringify(this.filterValues);
+    });
+    this.lastNameFilter.valueChanges.subscribe((lastName) => {
+      this.filterValues.lastName = lastName;
+      this.dataSource.filter = JSON.stringify(this.filterValues);
+    });
+  }
+
+  createFilter(): (data: any, filter: string) => boolean {
```
Comment (RomPavel) [inline src/app/models/hr-profile/hr-profile.component.ts:65]: dont use any

File: src/app/models/hr-profile/hr-profile.component.ts [hunk 2]
```diff
@@ -24,20 +39,50 @@ export class HrProfileComponent implements AfterViewInit {
...
+      data: {
+        id: { toString: () => string };
+        firstName: string;
+        lastName: { toString: () => string };
+      },
+      filter: string,
+    ): boolean {
+      const searchTerms = JSON.parse(filter);
+      return (
+        data.id.toString().toLowerCase().indexOf(searchTerms.id) !== -1 &&
```
Comment (RomPavel) [inline src/app/models/hr-profile/hr-profile.component.ts:76]: lets move this in a helper function isSubcstring(str, substr)

File: src/app/models/coach-profile/coach-profile-table/coach-profile-table.component.ts [hunk 3]
```diff
@@ -53,16 +71,41 @@ export class CoachProfileTableComponent implements AfterViewInit, OnChanges {
...
+        data.id.toString().toLowerCase().indexOf(searchTerms.id) !== -1 &&
+        data.level.toLowerCase().indexOf(searchTerms.level) !== -1 &&
+        data.date.toString().toLowerCase().indexOf(searchTerms.date) !== -1
+      );
+    };
+    return filterFunction;
+  }
+
+  onKey(event: any) {
+    this.searchQuery = event.target.value;
```
Comment (RomPavel) [inline src/app/models/coach-profile/coach-profile-table/coach-profile-table.component.ts:109]: why you need this

File: src/app/models/hr-profile/hr-profile.component.ts (hunk 1 above)
Comment (ExxZzyy) [reply src/app/models/hr-profile/hr-profile.component.ts:65]: For this
image

File: src/app/models/coach-profile/coach-profile-table/coach-profile-table.component.ts (hunk 3 above)
Comment (ExxZzyy) [reply src/app/models/coach-profile/coach-profile-table/coach-profile-table.component.ts:109]: For this
image

File: src/app/models/hr-profile/hr-profile.component.ts (hunk 1 above)
Comment (ExxZzyy) [reply src/app/models/hr-profile/hr-profile.component.ts]: Done efd3e3e0fa4cae085f6b2f0ea021d431433cef28

File: src/app/models/hr-profile/hr-profile.component.ts (hunk 2 above)
Comment (ExxZzyy) [reply src/app/models/hr-profile/hr-profile.component.ts]: Done efd3e3e0fa4cae085f6b2f0ea021d431433cef28

File: src/app/models/coach-profile/coach-profile-table/coach-profile-table.component.html [hunk 4]
```diff
@@ -1,26 +1,45 @@
 <div class="mat-elevation-z8">
-  <mat-form-field appearance="standard">
-    <mat-label>Filter</mat-label>
-    <input matInput (keyup)="applyFilter($event)" placeholder="Enter test number" #input />
-  </mat-form-field>
+  <!--  <mat-form-field appearance="standard">-->
```
Comment (RomPavel) [inline src/app/models/coach-profile/coach-profile-table/coach-profile-table.component.html:2]: do u need this?

File: src/app/models/coach-profile-editor/editor-table/editor-table.component.ts [hunk 5]
```diff
@@ -11,20 +12,36 @@ import { MatPaginator } from '@angular/material/paginator';
...
 
 import { MatDialog } from '@angular/material/dialog';
+import { FormControl } from '@angular/forms';
 import { CoachEditorTest } from '../../../../mocks/users-utils.mock';
 import { CoachEditorTabs } from '../../../../constants/data-constants';
 import { GrammarAddingEditingDialogComponent } from '../grammar-adding-editing-dialog/grammar-adding-editing-dialog.component';
 import { TopicAddingEditingDialogComponent } from '../topic-adding-editing-dialog/topic-adding-editing-dialog.component';
 import { EditListeningDialogComponent } from '../edit-listening-dialog/edit-listening-dialog.component';
 
+export function isSubstring(str: string | number, substr: string): boolean {
```
Comment (RomPavel) [inline src/app/models/coach-profile-editor/editor-table/editor-table.component.ts:22]: We have app/helpers folder for those

Comment (RomPavel) [reply src/app/models/coach-profile-editor/editor-table/editor-table.component.ts:22]: I think substr also can be turned to lower case

File: src/app/models/coach-profile/coach-profile-table/coach-profile-table.component.html (hunk 4 above)
Comment (ExxZzyy) [reply src/app/models/coach-profile/coach-profile-table/coach-profile-table.component.html]: No, deleted it 93038f15ef8ee456a78076e2c599b5f8db11070f

File: src/app/models/coach-profile-editor/editor-table/editor-table.component.ts (hunk 5 above)
Comment (ExxZzyy) [reply src/app/models/coach-profile-editor/editor-table/editor-table.component.ts]: Done 93038f15ef8ee456a78076e2c599b5f8db11070f

**Review by RomPavel — APPROVED** [review]

**Pull request merged**